pkg/util/backoff/backoff.go

```diff
@@ -36,21 +36,36 @@ type Policy struct {
 	MaxErrors int
 }
 
+// ServerlessDefaultBackoffFactor is the default value for the serverless backoff delay
+const ServerlessDefaultBackoffInterval = 100 * time.Millisecond
```

Comment:
this gives us roughly 20 attempts at shutdown, right?

Reply:
Yes, a bit less if it's not failing instantly

Reply:
Would it make sense to move this to a package owned by the serverless team?

Reply:
I'm not sure as for now we managed to never import a serverless package from a non-serverless package.
I could rename it as `DefaultConstantBackoffDuration`? WDYT?

Reply:
It's okay not to if it's hard. I think serverless name is better, since the 100ms retry interval would likely be too small in other contexts.

Reply:
As much as possible, the `ConstantBackoffPolicy` shouldn't be Serverless-aware. There isn't any reason for any of the backoff policy to know what environment they're used in.
That's why in the other PR I recommended a `time.Duration` as a parameter of the `NewConstantBackoffPolicy` func, because that's different for the piece of creating the backoff policy instance: this one could eventually be aware that it's for a serverless env.

L115 of the `http/destination.go` file, a test is already done through the config object to have the `Destination` having a different behaviour for Serverless environments (if you ask me, that's also not ideal 😅 but this one might be more complex to improve), this is a better place to use a value specifically for Serverless. And on top of that, that could make sense to also use the config for the value, this might help you troubleshoot/mitigate issues for customers in the future, being able to ask them to use different values without changing the layer version they're using.

In code, that would be either Solution 1:
  - removing the `ServerlessDefaultBackoffInterval` const
  - something along those line in `http/endpoint.go`:

```go
	if endpoint.Origin == config.ServerlessIntakeOrigin {
		shouldRetry = false
		policy = backoff.NewConstantBackoffPolicy(
			coreConfig.Datadog.GetDuration("serverless.constant_backoff_interval") // <- the proposed change
		)
	}
```

or at least eventually Solution 2:
  - moving the `ServerlessConstantBackoffInterval` const in `pkg/logs/config/config.go` which already has consts for Serverless. Not that I removed the `Default` in the const name since it can't be changed anyway.
  - in `http/endpoint.go`:

```go
	if endpoint.Origin == config.ServerlessIntakeOrigin {
		shouldRetry = false
		policy = backoff.NewConstantBackoffPolicy(
			config.ServerlessConstantBackoffInterval // <- the proposed change
		)
	}
```

Reply:
makes sense!  `ServerlessDefaultBackoffInterval` removed in favor of a `serverless.constant_backoff_interval` config